packages/react-core/src/components/TreeView/examples/TreeViewCheckbox.tsx

```diff
@@ -0,0 +1,191 @@
+      ]
+    },
+    {
+      name: 'Sources',
+      id: 'example3-Sources',
+      checkProps: { 'aria-label': 'sources-check', checked: false },
+      children: [
+        {
+          name: 'Application 4',
+          id: 'example3-App4',
+          checkProps: { 'aria-label': 'app-4-check', checked: false },
+          children: [
+            {
+              name: 'Settings',
+              id: 'example3-App4Settings',
+              checkProps: { 'aria-label': 'app-4-settings-check', checked: false }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      name: 'Really really really long folder name that overflows the container it is in',
+      id: 'example3-Long',
+      checkProps: { 'aria-label': 'long-check', checked: false },
+      children: [
+        { name: 'Application 5', id: 'example3-App5', checkProps: { 'aria-label': 'app-5-check', checked: false } }
+      ]
+    }
+  ];
+
+  const [checkedItems, setCheckedItems] = React.useState<TreeViewDataItem[]>([]);
+
+  const onCheck = (event: React.ChangeEvent, treeViewItem: TreeViewDataItem) => {
+    const checked = (event.target as HTMLInputElement).checked;
+
+    const checkedItemTree = options.map(opt => Object.assign({}, opt)).filter(item => filterItems(item, treeViewItem));
+    const flatCheckedItems = flattenTree(checkedItemTree);
+
+    setCheckedItems(
+      checked
+        ? checkedItems.concat(flatCheckedItems.filter(item => !checkedItems.some(i => i.id === item.id)))
+        : checkedItems.filter(item => !flatCheckedItems.some(i => i.id === item.id))
+    );
+  };
+
+  // Helper functions
+  const isChecked = (dataItem: TreeViewDataItem) => checkedItems.some(item => item.id === dataItem.id);
+  const areAllDescendantsChecked = (dataItem: TreeViewDataItem) =>
+    dataItem.children ? dataItem.children.every(child => areAllDescendantsChecked(child)) : isChecked(dataItem);
+  const areSomeDescendantsChecked = (dataItem: TreeViewDataItem) =>
+    dataItem.children ? dataItem.children.some(child => areSomeDescendantsChecked(child)) : isChecked(dataItem);
+
+  const flattenTree = (tree: TreeViewDataItem[]) => {
+    let result: TreeViewDataItem[] = [];
+    tree.forEach(item => {
+      result.push(item);
+      if (item.children) {
+        result = result.concat(flattenTree(item.children));
+      }
+    });
+    return result;
+  };
+
+  const mapTree = (item: TreeViewDataItem) => {
+    const hasCheck = areAllDescendantsChecked(item);
+    // Reset checked properties to be updated
+    if (item.checkProps) {
+      item.checkProps.checked = false;
+
+      if (hasCheck) {
+        item.checkProps.checked = true;
+      } else {
+        const hasPartialCheck = areSomeDescendantsChecked(item);
+        if (hasPartialCheck) {
+          item.checkProps.checked = null;
+        }
+      }
+
+      if (item.children) {
+        return {
+          ...item,
+          children: item.children.map(child => mapTree(child))
+        };
+      }
+    }
```

Comment:
Is this added conditional for `item.checkProps` because `checkProps` isn't a required prop the `TreeViewDataItem?`

Reply:
Yes